src/Arduino_LoRaConnectionHandler.cpp

```diff
@@ -0,0 +1,247 @@
+
+/******************************************************************************
+   CONSTANTS
+ ******************************************************************************/
+
+static const unsigned long NETWORK_CONNECTION_INTERVAL = 30000;   /*    NOT USED    */
+
+/******************************************************************************
+   CTOR/DTOR
+ ******************************************************************************/
+
+LoRaConnectionHandler::LoRaConnectionHandler(const char *_appeui, const char *_appkey, _lora_band band) :
+  appeui(_appeui),
+  appkey(_appkey),
+  band(band),
+  lastConnectionTickTime(millis()),
+  connectionTickTimeInterval(CHECK_INTERVAL_IDLE),
+  keepAlive(false),
+  _on_connect_event_callback(NULL),
+  _on_disconnect_event_callback(NULL),
+  _on_error_event_callback(NULL) {
+  netConnectionState = NetworkConnectionState::INIT;
+}
+
+/******************************************************************************
+   PUBLIC MEMBER FUNCTIONS
+ ******************************************************************************/
+
+void LoRaConnectionHandler::init() {
+}
+
+// INIT, CONNECTING, CONNECTED, DISCONNECTING, DISCONNECTED, CLOSED, ERROR
+void LoRaConnectionHandler::addCallback(NetworkConnectionEvent const event, OnNetworkEventCallback callback) {
+  switch (event) {
+    case NetworkConnectionEvent::CONNECTED:       _on_connect_event_callback       = callback; break;
+    case NetworkConnectionEvent::DISCONNECTED:    _on_disconnect_event_callback    = callback; break;
+    case NetworkConnectionEvent::ERROR:           _on_error_event_callback         = callback; break;
+    case NetworkConnectionEvent::INIT:                                                       ; break;
+    case NetworkConnectionEvent::CONNECTING:                                                 ; break;
+    case NetworkConnectionEvent::DISCONNECTING:                                              ; break;
+    case NetworkConnectionEvent::CLOSED:                                                     ; break;
+  }
+}
+
+void LoRaConnectionHandler::addConnectCallback(OnNetworkEventCallback callback) {
+  _on_connect_event_callback = callback;
+}
+void LoRaConnectionHandler::addDisconnectCallback(OnNetworkEventCallback callback) {
+  _on_disconnect_event_callback = callback;
+}
+void LoRaConnectionHandler::addErrorCallback(OnNetworkEventCallback callback) {
+  _on_error_event_callback = callback;
+}
+
+void LoRaConnectionHandler::execNetworkEventCallback(OnNetworkEventCallback & callback, void * callback_arg) {
+  if (callback) {
+    (*callback)(callback_arg);
+  }
+}
+
+unsigned long LoRaConnectionHandler::getTime() {
+  return 0;
+}
+
+int LoRaConnectionHandler::write(const uint8_t *buf, size_t size) {
+  int err;
+  modem.beginPacket();
+  modem.write(buf, size);
+  err = modem.endPacket(true);
+  /*Error manager according pr #68 of MKRWAN repo*/
+  if (err != size) {
+    switch (err) {
+      case -20: {
```

Comment:
Can we use constants for the error condes instead of naked integer values? Are they defined somewhere within the LoRa library?

Reply:
Unfortunately the error codes are not defined in the MKRWAN library. I added the error codes definition in the header of LoRaConnectionHandler

Reply:
:+1: